From 0b220f6e5e8fadd3e3203dc92cf15c80380a0770 Mon Sep 17 00:00:00 2001
From: Tim Foster <tim.foster@joyent.com>
Date: Mon, 30 Jul 2018 13:14:52 +0100
Subject: [PATCH] TRITON-554 sdc-dockerlogger build fails when cloned to
 anything other than 'dockerlogger' Reviewed by: Trent Mick <trentm@gmail.com>
 Approved by: Trent Mick <trentm@gmail.com>

---
 Makefile | 23 +++++++++++++++++------
 1 file changed, 17 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index e1dd2fb..2d7fb5c 100644
--- a/Makefile
+++ b/Makefile
@@ -5,13 +5,20 @@
 #
 
 #
-# Copyright (c) 2017, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 ELFEDIT=/usr/bin/elfedit
 GOPATH=$(PWD)/vendor
 TARGETS=dockerlogger dockerlogger.smartos
 
+_AWK := $(shell (which gawk >/dev/null && echo gawk) \
+	|| (which nawk >/dev/null && echo nawk) \
+	|| echo awk)
+BRANCH := $(shell git symbolic-ref HEAD | $(_AWK) -F/ '{print $$3}')
+
+DESTDIR ?= .
+
 ifeq ($(shell uname -s),Darwin)
 	TARGET = dockerlogger
 else
@@ -25,17 +32,15 @@ dockerlogger.smartos: dockerlogger
 	/usr/bin/elfedit -e 'ehdr:ei_osabi ELFOSABI_SOLARIS' dockerlogger dockerlogger.smartos
 
 dockerlogger: dockerlogger.go
-	GOPATH=$(GOPATH) go build
+	GOPATH=$(GOPATH) go build $<
 
 .PHONY: fmt
 fmt:
 	gofmt -w dockerlogger.go
 
 .PHONY: pkg
-pkg: dockerlogger.smartos
-	@[[ -n "$(BRANCH)" ]] || (echo "missing BRANCH="; exit 1)
-	@[[ -n "$(DESTDIR)" ]] || (echo "missing DESTDIR="; exit 1)
-	./tools/mk-shar -b $(BRANCH) -o $(DESTDIR)
+pkg: $(DESTDIR) dockerlogger.smartos
+	./tools/mk-shar -b "$(BRANCH)" -o $(DESTDIR)
 
 .PHONY: check
 check:
@@ -51,4 +56,10 @@ test: $(TARGET) node_modules/tape/bin/tape
 .PHONY: clean
 clean:
 	rm -f $(TARGETS)
+	rm -f $(DESTDIR)/dockerlogger-*.manifest
+	rm -f $(DESTDIR)/dockerlogger-*.md5sum
+	rm -f $(DESTDIR)/dockerlogger-*.sh
 	rm -rf build
+
+$(DESTDIR):
+	mkdir -p $@
-- 
2.21.0

